src/ZPublisher/tests/test_pubevents.py

```diff
@@ -159,8 +159,13 @@ def _succeed():
     return 'success'
 
 
+# Poor man's mock.
 class _Application(object):
-    pass
+    def __getattr__(self, name):
+        return self
+
+    def __call__(self, *args, **kwargs):
+        return self
```

Comment:
All the tests in `test_pubevents` using the publisher, which is:
```
testSuccess
testFailureReturn
testFailureException
testFailureConflict
```
These tests call `publish_module`. Now with my change, to explicitly load the application, the app returned via `get_module_info()` must be a wrapper. Thus supporting the `__call__` method. The return value of this `__call__` must have a `_p_jar` member which must have a `close()` method, therefore I added the `__getattr__` method.

All of these tests verify that the events inside the publisher are emitted, but do not care what the application really does. They use a custom request object with its own implementation of `traverse` to achieve that.